genshin/client/components/diary.py

```diff
@@ -90,28 +105,38 @@ async def get_diary(
         self,
         uid: typing.Optional[int] = None,
         *,
+        game: typing.Optional[types.Game] = None,
         month: typing.Optional[int] = None,
         lang: typing.Optional[str] = None,
-    ) -> models.Diary:
+    ) -> typing.Union[models.Diary, models.StarRailDiary]:
```

Comment:
This should probably use an `@overload` or be split into separate functions. Not sure if it's that useful so judge yourself.